src/esp/gfx/replay/Player.cpp

```diff
@@ -25,17 +55,22 @@ void Player::readKeyframesFromJsonDocument(const rapidjson::Document& d) {
 Keyframe Player::keyframeFromString(const std::string& keyframe) {
   Keyframe res;
   rapidjson::Document d;
-  d.Parse<0>(keyframe.c_str());
+  d.Parse<0>(keyframe.data(), keyframe.size());
   esp::io::readMember(d, "keyframe", res);
   return res;
 }
 
-Player::Player(const LoadAndCreateRenderAssetInstanceCallback&
-                   loadAndCreateRenderAssetInstanceCallback,
-               const ChangeLightSetupCallback& changeLightSetupCallback)
-    : loadAndCreateRenderAssetInstanceCallback(
-          loadAndCreateRenderAssetInstanceCallback),
-      changeLightSetupCallback(changeLightSetupCallback) {}
+Keyframe Player::keyframeFromStringUnwrapped(
+    const Cr::Containers::StringView keyframe) {
+  Keyframe res;
+  rapidjson::Document d;
+  d.Parse<0>(keyframe.data(), keyframe.size());
+  esp::io::fromJsonValue(d, res);
+  return res;
+}
```

Comment:
@eundersander I went ahead and added an optimized variant of the keyframe loading, see https://github.com/facebookresearch/habitat-sim/pull/1947/commits/136211fa611887516f4c0a49738c380b2ea54b24 for the full change. In particular:

- it takes a string view, because otherwise it'd have to allocate a `std::string` for every keyframe sliced out of the input file
- it expects it to contain directly the keyframe object, i.e. containing `"loads"` and `"creations"`, not a `"keyframe"
- and it passes an explicit size because the keyframe data can get rather huge and running `strlen()` on it again feels unnecessary

I kep the original API intact to avoid breaking any existing use. Is that fine or should we switch all code to use the new API?